### PR TITLE
chore: add topics and disposable token code snippets for dev docs

### PR DESCRIPTION
**Pull request by anitarua**
Addresses [#453](https://github.com/momentohq/dev-eco-issue-tracker/issues/453) by adding code snippet for `GenerateDisposableToken` that will be pulled into the dev docs. This snippet can be used [here](https://docs.momentohq.com/develop/authentication/tokens) and [here](https://docs.momentohq.com/develop/api-reference/auth#generatedisposabletoken-api).

Addresses [#459](https://github.com/momentohq/dev-eco-issue-tracker/issues/459) by adding code snippets for instantiating a topic client, publishing to a topic, and subscribing to a topic. These examples will be used [on this page](https://docs.momentohq.com/develop/api-reference/topics).

### CODE REVIEW

File: examples/DocExampleApis/Program.cs
Comment (pgautier404) [inline examples/DocExampleApis/Program.cs:163]: This is 100% the fault of the example script I threw together, but I think this approach of creating a custom scope with multiple permissions is possibly over-complicated for the API examples. The Javascript example uses the `DisposableTokenScopes` convenience methods to simplify this process and I think we should probably follow that example. So, for example, the selected code would omit the construction of the `scope` variable and use a predefined scope instead:

```
var tokenResponse = await client.GenerateDisposableTokenAsync(
    DisposableTokenScopes.CacheKeyReadWrite("squirrels", "mo"),
    ExpiresIn.Minutes(5)
);
```

The Javascript example shows off 4 of the predefined scopes, and I think it would be good to show multiple of the `DisposableTokenScopes` here too.

Sorry for causing the extra work and for the over-verbose example code. I'm going to either add a second disposable token example program using the predefined scopes or just add an example of each approach in the one program soon so users can study and compare both approaches, but for these docs I think the simplest approach is the best to show.

Comment (anitarua) [reply examples/DocExampleApis/Program.cs:163]: No worries! Easy enough to fix and better to be consistent with the existing examples for sure. I'll use predefined scopes to address the other 3 examples in the JS version as well.

File: examples/DocExampleApis/Program.cs
Comment (anitarua) [inline examples/DocExampleApis/Program.cs:178]: This one can be changed to remove `CacheSelector.ByName` once the additional predefined scope functions are released

Comment (pgautier404) [reply examples/DocExampleApis/Program.cs:178]: Ah, so we're missing an overload that takes `(string, TopicSelector)`. Good catch. Would you mind making a ticket to remind us to audit these functions and add signatures for `(string, XYZSelector)` please?

Comment (pgautier404) [reply examples/DocExampleApis/Program.cs:178]: Thank you for adding those!

**Comment by pgautier404** [comment]
This LGTM! I think @nand4011 should take a look at the Topic examples before you merge, and if he's good with it, :shipit:

File: examples/DocExampleApis/Program.cs
Comment (nand4011) [inline examples/DocExampleApis/Program.cs:228]: I don't think we need a break here after the exception. Looks like there are a couple of these.

File: examples/DocExampleApis/Program.cs
Comment (nand4011) [inline examples/DocExampleApis/Program.cs:150]: I don't normally comment about whitespace, but looking good is half the point of examples :-).

**Review by nand4011 — APPROVED** [review]

**Pull request merged**
